test/preview/components/preprocessors/test_text_document_splitter.py

```diff
@@ -0,0 +1,157 @@
+import pytest
+
+from haystack.preview import Document
+from haystack.preview.components.preprocessors import TextDocumentSplitter
+
+
+class TestTextDocumentSplitter:
+    @pytest.mark.unit
+    def test_non_text_document(self):
+        with pytest.raises(
+            ValueError, match="TextDocumentSplitter only works with text documents but document.text is None."
+        ):
+            splitter = TextDocumentSplitter()
+            splitter.run(documents=[Document()])
+
+    @pytest.mark.unit
+    def test_single_doc(self):
+        with pytest.raises(TypeError, match="TextDocumentSplitter expects a List of Documents as input."):
+            splitter = TextDocumentSplitter()
+            splitter.run(documents=Document())
+
+    @pytest.mark.unit
+    def test_empty_list(self):
+        with pytest.raises(TypeError, match="TextDocumentSplitter expects a List of Documents as input."):
+            splitter = TextDocumentSplitter()
+            splitter.run(documents=[])
```

Comment:
Oh that's a nice point. Should we make the error more specific by distinguishing empty lists from Nones and other types? Not blocking, just an idea.

Reply:
I'd leave it as is for now. If we want to give error messages on that level of detail, let's do that in the same way across all components that expect a List of Document as input. All the embedders, etc.